cypress/e2e/base.spec.ts

```diff
@@ -20,4 +20,94 @@ describe('Visit directory page and verify users exists', () => {
   });
 });
 
+describe('directory infinite scroll', () => {
+  const PAGE_LIMIT = 20;
+  let totalPages = 0;
+  function checkInfiniteScroll(pages: number, totalPages: number) {
+    for (let i = 1; i <= pages; i++) {
+      cy.get('[data-cy="user-container"]').children().as('ulChildren');
+      i < totalPages
+        ? cy.get('@ulChildren').should('have.length', PAGE_LIMIT * i)
+        : // last page not guaranteed to be full
+        cy.get('@ulChildren').should('have.length.within', PAGE_LIMIT * (i - 1), PAGE_LIMIT * i);
+
+      cy.intercept('GET', '/api/users?page=*', req => {
+        req.on('response', res => res.setDelay(500));
+      }).as('getUsers');
+      cy.scrollTo('bottom');
+
+      // last page won't have a spinner
+      if (i === totalPages) break;
+      // check spinner happened & removed
+      cy.get('[data-cy="loading"]'); //check if exists
+      cy.get('[data-cy="loading"]').should('not.exist'); // removed after data loaded
+    }
+  }
+
+  before(() => {
+    // get total pages for tests
+    cy.request('http://localhost:3000/api/users').then(response => {
+      totalPages = response.body.totalPages;
+    });
+
+    // clear the db then fill with 100 users
+    // User.destroy({ truncate: true })
+    //   .then(() => {
+    //     runSeedCommand();
+    //   })
+    //   .catch(err => {
+    //     console.error('Error deleting existing users:', err);
+    //   });
```

Comment:
I couldn't get Cypress to accept any imports so this errors. But the intention is I believe we should clear the DB before seeding the DB with 100 users so we know what to expect.